### ISSUE TITLE
Numpy 2.0 update causing issues with other software

### ISSUE THREAD

**Issue by Josh-XT** [explosion/spaCy#13669]
Forcing numpy to 2.0+ forces breaking changes on other software. Unless it is absolutely required to use 2.0+, can this be reverted?

https://github.com/explosion/spaCy/commit/184e508d9c8db6fc89c60e5ec8e94324817259c9

**Comment by rphellan** [comment]
Yes, also numpy 2.0.0 requires python 3.9. Then, the line `numpy>=2.0.0,<2.1.0; python_version < '3.9'"` leads to a bug.

**Comment by Darshan2104** [comment]
Yes it is creating issue. I am facing issue while working with ```keyphrase-vectorizers==0.0.10``` where it is using spacy version >=3.0.0 and it is installing ```spacy==3.8.2``` i'm using python3.8 with ```numpy==1.24.3``` which is compatable but spacy3.8.2 is looking for numpy2.0+ and that is not compatable with python3.8.

Does anyone has any idea which spacy (>=3.0.1) is compatable and better with python3.8?

**Comment by lsmith77** [comment]
adding fasttext to the list of libs not compatible (and unfortunately also not maintained). we have found it to be the best language detection for shorter text.

**Comment by simon-ging** [comment]
@lsmith77 off-topic but I uploaded a numpy2 version of fasttext, because I ran into the same problem, and it was just a one line fix. see [https://pypi.org/project/fasttext-numpy2/](https://pypi.org/project/fasttext-numpy2/)

**Comment by honnibal** [comment]
There's no way to support both numpy v1 and numpy v2 in the same release. You'll need to pin your spaCy dependency to have an upper-bound that uses numpy v1.

**Comment by honnibal** [comment]
This should be resolved now. numpy v2 is specified as a build-time dependency, while runtime is compatible with both.
